app/assets/javascripts/discourse/app/components/modal/revise-and-reject-post-reviewable.js

```diff
@@ -0,0 +1,52 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { isEmpty } from "@ember/utils";
+
+const OTHER_REASON = "Other...";
+
+export default class ReviseAndRejectPostReviewable extends Component {
+  @service siteSettings;
+
+  @tracked reason = null;
+  @tracked customReason = null;
+  @tracked feedback = null;
+  @tracked submitting = false;
+
+  get configuredReasons() {
+    return this.siteSettings.reviewable_revision_reasons
+      .split("|")
+      .concat([OTHER_REASON]);
+  }
+
+  get showCustomReason() {
+    return this.reason === OTHER_REASON;
+  }
+
+  get sendPMDisabled() {
+    return (
+      isEmpty(this.reason) ||
+      (this.reason === OTHER_REASON && isEmpty(this.customReason)) ||
+      this.submitting
+    );
+  }
+
+  @action
+  rejectAndSendPM() {
+    this.submitting = true;
+    this.args.model
+      .performConfirmed(this.args.model.action, {
+        revise_reason: this.reason,
+        revise_custom_reason: this.customReason,
+        revise_feedback: this.feedback,
+      })
+      .then(() => {
+        this.submitting = false;
+        this.args.closeModal();
+      })
+      .finally(() => {
+        this.submitting = false;
+      });
+  }
```

Comment:
Could be nice to use async/await with try/catch for new code

Reply:
Oh yep sure, will change :+1: